Web/Edubase.Web.UIUnitTests/Areas/ViewRulesHandlers/GroupEditorViewModelRulesHandlerTests.cs

```diff
@@ -0,0 +1,253 @@
+            mockPrincipal.Setup(x => x.IsInRole(role)).Returns(true);
+            var viewModel = new GroupEditorViewModel()
+            {
+                GroupTypeId = (int) groupType,
+                StatusId = (int) statusId
+            };
+
+            var result = GroupEditorViewModelRulesHandler.SetEditPermissions(viewModel, mockPrincipal.Object);
+
+            Assert.True(result.CanUserCloseAndMarkAsCreatedInError);
+        }
+
+        [Theory]
+        [InlineData(eLookupGroupType.MultiacademyTrust, eLookupGroupStatus.CreatedInError, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.MultiacademyTrust, eLookupGroupStatus.Closed, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.SingleacademyTrust, eLookupGroupStatus.CreatedInError, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.SingleacademyTrust, eLookupGroupStatus.Closed, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.SchoolSponsor, eLookupGroupStatus.CreatedInError, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.SchoolSponsor, eLookupGroupStatus.Closed, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.Federation, eLookupGroupStatus.CreatedInError, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.Federation, eLookupGroupStatus.Closed, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.SecureSingleAcademyTrust, eLookupGroupStatus.CreatedInError, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.SecureSingleAcademyTrust, eLookupGroupStatus.Closed, EdubaseRoles.ROLE_BACKOFFICE)]
+        [InlineData(eLookupGroupType.SecureSingleAcademyTrust, eLookupGroupStatus.CreatedInError, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.SecureSingleAcademyTrust, eLookupGroupStatus.Closed, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.SecureSingleAcademyTrust, eLookupGroupStatus.Open, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.MultiacademyTrust, eLookupGroupStatus.CreatedInError, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.MultiacademyTrust, eLookupGroupStatus.Closed, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.SingleacademyTrust, eLookupGroupStatus.CreatedInError, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.SingleacademyTrust, eLookupGroupStatus.Closed, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.SchoolSponsor, eLookupGroupStatus.CreatedInError, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.SchoolSponsor, eLookupGroupStatus.Closed, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.Federation, eLookupGroupStatus.CreatedInError, EdubaseRoles.YCS)]
+        [InlineData(eLookupGroupType.Federation, eLookupGroupStatus.Closed, EdubaseRoles.YCS)]
+        public void SetEditPermissions_GroupViewModel_CanUserCloseAndMarkAsCreatedInError_False(eLookupGroupType groupType, eLookupGroupStatus statusId, string role)
```

Comment:
It is hard to grasp if these test inputs are "complete enough" without context / description about the data.

A hard problem to solve, and I'm not sure how helpful it is to point this out - sorry!

Personally I wouldn't object if these tests are added as-is given that it's an improvement on what is currently in the repo.

Reply:
@CathLass any thoughts?

Reply:
I had a look at this and it is checking for false to be returned.  The two statuses are the only ones that return false.
It looks like it contains all of the possible combinations that are relevant.